src/parser/mod.rs

```diff
@@ -1078,6 +1078,9 @@ impl<'a> Parser<'a> {
                     let expr = self.parse_subexpr(Self::PLUS_MINUS_PREC)?;
                     Ok(Expr::Prior(Box::new(expr)))
                 }
+                Keyword::MAP if self.peek_token() == Token::LBrace => {
```

Comment:
Should this also check `dialect.support_map_literal_syntax()`?

Reply:
Sounds great. Thanks